test/logic/stake_logic_test.rb

```diff
@@ -257,11 +257,55 @@ def setup
       }
     )
     p = Pipeline.new(200, @initial_payload)
-                .then(&calculate_apy)
+                .then(&calculate_apy_for_accounts)
     
     acc.reload
-
     assert_equal 200, p.code
     assert_equal 12.5502, acc.apy
   end
+
+  test "calculate_apy_for_pools should return correct average apy" do
```

Comment:
This one test is enough for this feature? There are couple of `if` statements in this method, maybe it's worth to test them too.